solr/core/src/java/org/apache/solr/cli/CreateTool.java

```diff
@@ -131,7 +134,7 @@ public void runImpl(CommandLine cli) throws Exception {
 
   protected void createCore(CommandLine cli, SolrClient solrClient) throws Exception {
     String coreName = cli.getOptionValue("name");
-    String solrUrl = cli.getOptionValue("solrUrl", SolrCLI.getDefaultSolrUrl());
+    String solrUrl = cli.getOptionValue("solrUrl", SolrCLI.getDefaultSolrUrl(cli));
```

Comment:
Out of curiosity, if `SolrCLI.getDefaultSolrUrl(cli))` always takes a cli, should we make it be `cli.getDefaultSolrUrl()` instead?